lib/internal/modules/cjs/loader.js

```diff
@@ -1529,6 +1560,113 @@ function getMaybeCachedSource(mod, filename) {
   return content;
 }
 
+/**
+ * Resolve and evaluate it synchronously as ESM if it's ESM.
+ * @param {Module} mod CJS module instance
+ * @param {string} filename Absolute path of the file.
+ */
+function loadMTS(mod, filename) {
+  if (StringPrototypeIncludes(filename, 'node_modules')) {
```

Comment:
Technically I think this would also cause `require('node_modules.ts')` to throw. Maybe that’s too edge case-y to care about? But the solution would be to get the `basename` of `filename` and do the `includes` check on that.

Reply:
fixed and checked with a test